tracker/table.go

```diff
@@ -397,6 +400,41 @@ func (e *Order) Title(x int) (title string) {
 	return
 }
 
+func (m *Order) instrumentListFor(trackIndex int) (int, int, error) {
+	track := m.d.Song.Score.Tracks[trackIndex]
+	firstVoice := m.d.Song.Score.FirstVoiceForTrack(trackIndex)
+	lastVoice := firstVoice + track.NumVoices - 1
+	firstIndex, err1 := m.d.Song.Patch.InstrumentForVoice(firstVoice)
+	if err1 != nil {
+		return trackIndex, trackIndex, err1
+	}
+	lastIndex, err2 := m.d.Song.Patch.InstrumentForVoice(lastVoice)
+	if err2 != nil {
+		return trackIndex, trackIndex, err2
+	}
+	return firstIndex, lastIndex, nil
+}
+
+func (m *Order) TrackIndicesForCurrentInstrument() []int {
```

Comment:
Please use iterators instead of returning slices (which should become easy once AllTracksWithSameInstrument has an iterator signature as well)